source/Windows.Devices.I2c/I2cController.cs

```diff
@@ -38,10 +41,7 @@ internal static Hashtable DeviceCollection
                 {
                     lock (_syncLock)
                     {
-                        if (s_deviceCollection == null)
-                        {
-                            s_deviceCollection = new Hashtable();
-                        }
+                        s_deviceCollection = new Hashtable();
```

Comment:
You need to keep the lazy init pattern here. After entering the lock you have to check again if the instance is still null.
In case "this" was the waiting thread, the object could already have been created elsewhere.